test/contain_overlap_test.rb

```diff
@@ -46,4 +46,158 @@ def test_float
   end
+
+  def test_date
+    assert_equal false, MultiRange.new([Date.new(2001,2,3)..Date.new(2001,3,10), Date.new(2002,2,3)..Date.new(2002,3,10)]).contain_overlaps?
+    assert_equal true, MultiRange.new([Date.new(2001,2,3)..Date.new(2002,3,10), Date.new(2002,3,3)..Date.new(2002,3,10)]).contain_overlaps?
+    assert_equal true, MultiRange.new([Date.new(2001,2,3)..Date.new(2002,3,10), Date.new(2002,3,4)..Date.new(2002,3,10)]).contain_overlaps?
+  end
+
+  def test_time_overlaps
+    assert_equal false, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1),
+                     Time.at(3)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+
+    assert_equal false, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1),
+                     Time.at(2)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+
+    assert_equal true, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1),
+                     Time.at(1)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+
+    assert_equal true, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1, 500, :nsec),
+                     Time.at(1,999,:nsec)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+
+    assert_equal true, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1, 500, :nsec),
+                     Time.at(1,1499,:nsec)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+
+    assert_equal false, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1, 499, :nsec),
+                     Time.at(1,999,:nsec)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+
+    assert_equal false, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1),
+                     Time.at(1,500,:nsec)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+
+    assert_equal false, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1),
+                     Time.at(1,1499,:nsec)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+
+    assert_equal false, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1),
+                     Time.at(1,1,:microsecond)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+
+    assert_equal false, MultiRange.new(
+                   [
+                     Time.at(0)..Time.at(1),
+                     Time.at(1,1,:millisecond)..Time.at(4)
+                   ]
+                 ).contain_overlaps?
+  end
+
+  def test_active_support_time_with_timezone_overlaps
+    require 'active_support/time_with_zone'
+    require 'active_support/time'
+    Time.zone = 'Eastern Time (US & Canada)'
+
```

Comment:
The [test case fails](https://github.com/khiav223577/multi_range/actions/runs/4576298812/jobs/8238817972?pr=28) at Line 131. It may need to add `require "active_support/isolated_execution_state"` here?

The error message is:
```
ContainOverlapTest#test_active_support_time_with_timezone_overlaps:
NameError: uninitialized constant ActiveSupport::IsolatedExecutionState
    /home/runner/work/multi_range/multi_range/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.4.3/lib/active_support/core_ext/time/zones.rb:42:in `zone='
    /home/runner/work/multi_range/multi_range/test/contain_overlap_test.rb:131:in `test_active_support_time_with_timezone_overlaps'
```